File: hw2/RaceMonitor_Java/test/Messages/RegistrationUpdateTest.java
package Messages;

import Racedata.AthleteRaceStatus;
import org.junit.Test;

import java.time.LocalDateTime;

import static org.junit.Assert.*;

public class RegistrationUpdateTest {
    @Test
    public void create() throws Exception {
        RegistrationUpdate rUpdate = (RegistrationUpdate) AthleteUpdate.Create("Registered,14,8/15/2017 7:02:05 AM,Jane,Jones,F,16");
        assertNotNull(rUpdate);

        assertEquals(AthleteRaceStatus.Registered, rUpdate.getUpdateType());
        assertEquals(14, rUpdate.getBibNumber());
        assertEquals(LocalDateTime.of(2017,8,15,7, 2, 5), rUpdate.getTimestamp());
        assertEquals("Jane", rUpdate.getFirstName());
        assertEquals("Jones", rUpdate.getLastName());
        assertEquals("F", rUpdate.getGender());
        assertEquals(16, rUpdate.getAge());

        try {
            AthleteUpdate msg = AthleteUpdate.Create("Registered,14,8/15/2017 7:02:05 AM,Jane,Jones,F");
            fail("Exception expected");
        } catch (Exception e) {}

        try {
            AthleteUpdate msg = AthleteUpdate.Create("Registered,14,8/15/2017 7:02:05 AM,Jane,Jones");
            fail("Exception expected");
        } catch (Exception e) {}

        try {
            AthleteUpdate msg = AthleteUpdate.Create("Registered,14,8/15/2017 7:02:05 AM,Jane");
            fail("Exception expected");
        } catch (Exception e) {}

        try {
            AthleteUpdate msg = AthleteUpdate.Create("Registered,14,8/15/2017 7:02:05 AM");
            fail("Exception expected");
        } catch (Exception e) {}

        try {
            AthleteUpdate msg = AthleteUpdate.Create("Registered,14");
            fail("Exception expected");
        } catch (Exception e) {}

        try {
            AthleteUpdate msg = AthleteUpdate.Create("Registered");
            fail("Exception expected");
        } catch (Exception e) {}

        try {
            AthleteUpdate msg = AthleteUpdate.Create("Registered,14,8/15/2017 7:02:05 AM,Jane,Jones,F,16,bad");
            fail("Exception expected");
        } catch (Exception e) {}

    }

    @Test
    public void testToString() throws Exception {
        RegistrationUpdate rUpdate = new RegistrationUpdate();
        rUpdate.setBibNumber(14);
        rUpdate.setTimestamp(LocalDateTime.of(2017,8,15,7,2,5));
        rUpdate.setFirstName("Jane");
        rUpdate.setLastName("Jones");
        rUpdate.setGender("F");
        rUpdate.setAge(16);

        assertEquals("Registered,14,8/15/2017 7:02:05 AM,Jane,Jones,F,16", rUpdate.toString());
    }

    @Test
    public void testSettersAndGetters() throws Exception {
        RegistrationUpdate rUpdate = new RegistrationUpdate();
        rUpdate.setBibNumber(14);
        rUpdate.setTimestamp(LocalDateTime.of(2017,8,15,7,2,5));
        rUpdate.setFirstName("Jane");
        rUpdate.setLastName("Jones");
        rUpdate.setGender("F");
        rUpdate.setAge(16);

        assertEquals(AthleteRaceStatus.Registered, rUpdate.getUpdateType());
        assertEquals(14, rUpdate.getBibNumber());
        assertEquals(LocalDateTime.of(2017,8,15,7, 2, 5), rUpdate.getTimestamp());
        assertEquals("Jane", rUpdate.getFirstName());
        assertEquals("Jones", rUpdate.getLastName());
        assertEquals("F", rUpdate.getGender());
        assertEquals(16, rUpdate.getAge());
    }
}

File: hw2/RaceMonitor_CSharp/RaceDataTesting/Messages/RegistrationUpdateTest.cs
using System;
using Microsoft.VisualStudio.TestTools.UnitTesting;

using RaceData;
using RaceData.Messages;

namespace RaceDataTesting.Messages
{
    [TestClass]
    public class RegistrationUpdateTest
    {
        [TestMethod]
        public void RegistrationUpdate_Create()
        {
            RegistrationUpdate rUpdate = (RegistrationUpdate)AthleteUpdate.Create("Registered,14,8/15/2017 7:02:05 AM,Jane,Jones,F,16");
            Assert.IsNotNull(rUpdate);

            Assert.AreEqual(AthleteRaceStatus.Registered, rUpdate.UpdateType);
            Assert.AreEqual(14, rUpdate.BibNumber);
            Assert.AreEqual(new DateTime(2017, 8, 15, 7, 2, 5), rUpdate.Timestamp);
            Assert.AreEqual("Jane", rUpdate.FirstName);
            Assert.AreEqual("Jones", rUpdate.LastName);
            Assert.AreEqual("F", rUpdate.Gender);
            Assert.AreEqual(16, rUpdate.Age);

            try
            {
                var msg = AthleteUpdate.Create("Registered,14,8/15/2017 7:02:05 AM,Jane,Jones,F");
                Assert.Fail($"Exception expected when creating: msg={msg}");
            }
            catch { /* ignore */ }

            try
            {
                var msg = AthleteUpdate.Create("Registered,14,8/15/2017 7:02:05 AM,Jane,Jones");
                Assert.Fail($"Exception expected when creating: msg={msg}");
            }
            catch { /* ignore */ }

            try
            {
                var msg = AthleteUpdate.Create("Registered,14,8/15/2017 7:02:05 AM,Jane");
                Assert.Fail($"Exception expected when creating: msg={msg}");
            }
            catch { /* ignore */ }

            try
            {
                var msg = AthleteUpdate.Create("Registered,14,8/15/2017 7:02:05 AM");
                Assert.Fail($"Exception expected when creating: msg={msg}");
            }
            catch { /* ignore */ }

            try
            {
                var msg = AthleteUpdate.Create("Registered,14");
                Assert.Fail($"Exception expected when creating: msg={msg}");
            }
            catch { /* ignore */ }

            try
            {
                var msg = AthleteUpdate.Create("Registered");
                Assert.Fail($"Exception expected when creating: msg={msg}");
            }
            catch { /* ignore */ }

            try
            {
                var msg = AthleteUpdate.Create("Registered,14,8/15/2017 7:02:05 AM,Jane,Jones,F,16,bad");
                Assert.Fail($"Exception expected when creating: msg={msg}");
            }
            catch { /* ignore */ }

        }

        [TestMethod]
        public void RegistrationUpdate_ToString()
        {
            RegistrationUpdate rUpdate = new RegistrationUpdate
            {
                BibNumber = 14,
                Timestamp = new DateTime(2017, 8, 15, 7, 2, 5),
                FirstName = "Jane",
                LastName = "Jones",
                Gender = "F",
                Age = 16
            };

            Assert.AreEqual("Registered,14,8/15/2017 7:02:05 AM,Jane,Jones,F,16", rUpdate.ToString());
        }


        [TestMethod]
        public void RegistrationUpdate_GetterAndSetters()
        {
            RegistrationUpdate rUpdate = new RegistrationUpdate
            {
                BibNumber = 14,
                Timestamp = new DateTime(2017, 8, 15, 7, 2, 5),
                FirstName = "Jane",
                LastName = "Jones",
                Gender = "F",
                Age = 16
            };

            Assert.AreEqual(AthleteRaceStatus.Registered, rUpdate.UpdateType);
            Assert.AreEqual(14, rUpdate.BibNumber);
            Assert.AreEqual(new DateTime(2017, 8, 15, 7, 2, 5), rUpdate.Timestamp);
            Assert.AreEqual("Jane", rUpdate.FirstName);
            Assert.AreEqual("Jones", rUpdate.LastName);
            Assert.AreEqual("F", rUpdate.Gender);
            Assert.AreEqual(16, rUpdate.Age);

        }
    }
}
